google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/stream.rb

```diff
@@ -53,6 +53,15 @@ def initialize subscriber
             @push_thread_pool = Concurrent::FixedThreadPool.new \
               subscriber.push_threads
 
+            @stream_keepalive_task = Concurrent::TimerTask.new(
+              execution_interval: 30
+            ) do
+              # push empty request every 30 seconds to keep stream alive
+              unless inventory.empty?
```

Comment:
@blowmage Why this conditional? Why not always send and keep open?

Reply:
I don't know. The directions we got were to keep the stream alive with empty requests as long as there were outstanding messages on the stream.

Reply:
We don't want to keep the stream open artificially if we're not using it.